rfcs/rfc-002-consensus.md

```diff
@@ -75,48 +82,101 @@ When the `Slash` vote is processed, validators should lookup `PoH hash` at `PoH
+2. I will not vote on a branch that doesn't contain this message for at least `N` counts, or until `PoH tick count` + `N` is reached by the PoH stream.
 3. I will not vote for any other branch below `PoH count`.
     a. if there are other votes not present in this PoH history the validator may need to `cancel` them before creating this vote.
 
 ## Leader Seed Generation
 
 Leader selection is decided via a random seed.  The process is as follows:
 
-1. Periodically at a specific `PoH count` select the first vote signatures that create a supermajority from the previous round.
-2. append them together
-3. hash the string for `N` counts via a similar process as PoH itself.
-4. The resulting hash is the random seed for `M` counts, where M > N
+1. Periodically at a specific `PoH tick count` select the first vote signatures that create a supermajority from the previous voting round.
+2. Append them together.
+3. Hash the string for `N` counts via a similar process as PoH itself.
+4. The resulting hash is the random seed for `M` counts, i.e. `M` voting rounds, i.e. `M` leader periods, where M > N
 
 ## Leader Ranking and Rotation
 
-Leader's transmit for a count of `T`.  When `T` is reached all the validators should switch to the next ranked leader.  To rank leaders, the supermajority + `M` nodes are shuffled with the using the above calculated random seed.
+    a. The leader is chosen via a random seed generated from stake weights and votes (the leader schedule)
+    b. The leader is rotated at every voting round, accoding to the leader schedule
+    c. The schedule is applicable for `M` voting rounds
+
+Leader's transmit for a count of `T` PoH ticks.  When `T` is reached all the validators should switch to the next ranked leader.  To rank leaders, the supermajority + `M` nodes are shuffled using the above calculated random seed.
+
+All `T` ticks must be observed from the current leader for that part of PoH to be accepted by the network.  If `T` ticks (and any intervening transactions) are not observed, the network optimistically fills in the `T` ticks, and continues with PoH from the next leader.
+
+## Partitions, Forks
+
+Forks may arise at PoH tick counts that correspond to leader rotations, i.e. voting points.
+
+A fork arises when a validator observes a timeout for the designated leader during that leader's period.  The validator speculatively rolls back to the beginning of that leader's period and inserts a fork of empty ticks for the period.  These empty ticks are generated by all nodes in the network at a network-specified rate for hashes/per/tick `Z`.
+
+There are only two possible versions of the PoH during a voting period: PoH with `T` ticks and entries generated by the current leader, or PoH with just ticks.  The "just ticks" version of the PoH can be thought of as a virtual ledger, one that all nodes in the network can derive from the last tick in the previous period.
```

Comment:
there could be more forks, basically any transactional fork that wasn't finalized by supermajority could produce a different fork